akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -136,12 +137,37 @@ private[akka] class Association(
               quarantine(reason = s"Due to overflow of control queue, size [$controlQueueSize]")
           }
         case _ ⇒
-          queue.offer(Send(message, senderOption, recipient, None))
+          val send = Send(message, senderOption, recipient, None)
+          if (!largeMessageChannelEnabled || !isLargeMessageDestination(recipient)) {
+            queue.offer(send)
+          } else {
+            largeQueue.offer(send)
+          }
```

Comment:
Did it like this to get the most-commonly selected branch first. But I agree, the negations makes it more tricky than necessary to read.